disco_aws_automation/disco_rds.py

```diff
@@ -170,80 +381,142 @@ def update_cluster(self, database_name):
+        rds.update_cluster()
+
+    def get_subnet_ids(self):
+        """ Get a list of subnet ids for the vpc"""
+        subnets = self.vpc.get_all_subnets()
+        subnet_ids = []
+        for subnet in subnets:
+            tags = tag2dict(subnet['Tags'])
+            if tags['meta_network'] == 'intranet':
+                subnet_ids.append(str(subnet['SubnetId']))
+        return subnet_ids
 
-    def update_all_clusters_in_vpc(self):
+    def update_all_clusters_in_vpc(self, parallel=True):
         """
         Updates every RDS instance in the current VPC to match the configuration
         """
         vpc_prefix = self.vpc_name + '-'
         sections = [section for section in self.config_rds.sections()
                     if section.startswith(vpc_prefix)]
         logging.debug("The following RDS clusters will be updated: %s", ", ".join(sections))
+        rds_security_group_id = self.get_rds_security_group_id()
+        subnet_ids = self.get_subnet_ids()
+        rds_list = []
+
         for section in sections:
             # the section names are database identifiers
             database_identifier = section
-            self.update_cluster(self._get_database_name(self.vpc_name, database_identifier))
-
-    def recreate_db_subnet_group(self, db_subnet_group_name):
-        """
-        Creates the DB Subnet Group. If it exists already, drops it and recreates it.
-        DB subnet groups must contain at least one subnet in at least two AZs in the region.
-
-        @db_subnet_group_name: String. The name for the DB subnet group.
-                               This value is stored as a lowercase string.
-        """
-        try:
-            self.client.delete_db_subnet_group(DBSubnetGroupName=db_subnet_group_name)
-        except Exception as err:
-            logging.debug("Not deleting subnet group '%s': %s", db_subnet_group_name, repr(err))
-
-        db_subnet_group_description = 'Subnet Group for VPC {0}'.format(self.vpc_name)
-        subnets = self.vpc.get_all_subnets()
-        subnet_ids = []
-        for subnet in subnets:
-            tags = tag2dict(subnet['Tags'])
-            if tags['meta_network'] == 'intranet':
-                subnet_ids.append(str(subnet['SubnetId']))
-
-        self.client.create_db_subnet_group(DBSubnetGroupName=db_subnet_group_name,
-                                           DBSubnetGroupDescription=db_subnet_group_description,
-                                           SubnetIds=subnet_ids)
+            rds = RDS(self.vpc_name,
+                      database_identifier,
+                      rds_security_group_id,
+                      subnet_ids,
+                      self.domain_name)
+            rds_list.append(rds)
+
+        # For Sequential execution
+        if not parallel:
+            for rds in rds_list:
+                rds.update_cluster()
+            return
+
+        # Running in parallel
+        for rds in rds_list:
+            rds.start()
```

Comment:
nit: is it ok to call `.join()` right after `.start()`? We could avoid an extra for loop if yes.